src/software/embedded/services/motor.cpp

```diff
@@ -440,37 +431,78 @@ TbotsProto::MotorStatus MotorService::poll(const TbotsProto::MotorControl& motor
     // Get current wheel electical RPM (don't account for pole pairs). We will use these
     // for robot status feedback We assume the motors have ramped to the expected RPM from
     // the previous iteration.
-    double front_right_velocity =
-        static_cast<double>(tmc4671ReadThenWriteValue(
-            FRONT_RIGHT_MOTOR_CHIP_SELECT, TMC4671_PID_VELOCITY_ACTUAL,
-            TMC4671_PID_VELOCITY_TARGET, front_right_target_rpm)) *
-        MECHANICAL_MPS_PER_ELECTRICAL_RPM;
-    double front_left_velocity =
-        static_cast<double>(tmc4671ReadThenWriteValue(
-            FRONT_LEFT_MOTOR_CHIP_SELECT, TMC4671_PID_VELOCITY_ACTUAL,
-            TMC4671_PID_VELOCITY_TARGET, front_left_target_rpm)) *
-        MECHANICAL_MPS_PER_ELECTRICAL_RPM;
-    double back_right_velocity =
-        static_cast<double>(tmc4671ReadThenWriteValue(
-            BACK_RIGHT_MOTOR_CHIP_SELECT, TMC4671_PID_VELOCITY_ACTUAL,
-            TMC4671_PID_VELOCITY_TARGET, back_right_target_rpm)) *
-        MECHANICAL_MPS_PER_ELECTRICAL_RPM;
-    double back_left_velocity =
-        static_cast<double>(tmc4671ReadThenWriteValue(
-            BACK_LEFT_MOTOR_CHIP_SELECT, TMC4671_PID_VELOCITY_ACTUAL,
-            TMC4671_PID_VELOCITY_TARGET, back_left_target_rpm)) *
-        MECHANICAL_MPS_PER_ELECTRICAL_RPM;
-    double dribbler_rpm = static_cast<double>(
-        tmc4671ReadThenWriteValue(DRIBBLER_MOTOR_CHIP_SELECT, TMC4671_PID_VELOCITY_ACTUAL,
-                                  TMC4671_PID_VELOCITY_TARGET, dribbler_ramp_rpm_));
+    double front_right_velocity;
+    double front_left_velocity;
+    double back_right_velocity;
+    double back_left_velocity;
+    double dribbler_rpm;
 
+    if (motorInEnabledList(FRONT_RIGHT_MOTOR_CHIP_SELECT))
+    {
+        front_right_velocity =
+            static_cast<double>(tmc4671ReadThenWriteValue(
+                FRONT_RIGHT_MOTOR_CHIP_SELECT, TMC4671_PID_VELOCITY_ACTUAL,
+                TMC4671_PID_VELOCITY_TARGET, front_right_target_rpm)) *
+            MECHANICAL_MPS_PER_ELECTRICAL_RPM;
+    }
+    else
+    {
+        // TODO #3424: When a motor is disabled use three-wheel velocity calculations.
+        front_right_velocity = 0;
+    }
```

Comment:
consider making a helper function called `writeEnabledMotor`. So, this could look like:

`front_right_velocity = writeEnabledMotor(FRONT_RIGHT_MOTOR_CHIP_SELECT, front_right_target_rpm)`